traceevents/src/main/java/com/tomtom/kotlin/traceevents/Tracer.kt

```diff
@@ -524,15 +560,22 @@ class Tracer private constructor(
         /**
          * Remove a trace event consumer. The event processor should not be stopped if there are
          * no consumers left, because it holds the event queue and you may lose events then.
+         *
+         * @param traceEventConsumer Trace event consumer to remove.
+         * @param contextRegex If this is `null`, all trace event consumers for the same tracers will be removed.
+         * Otherwise only the specific trace event consumer for that regular expression is removed.
          */
-        fun removeTraceEventConsumer(traceEventConsumer: TraceEventConsumer) {
-            traceEventConsumers.remove(traceEventConsumer)
+        fun removeTraceEventConsumer(traceEventConsumer: TraceEventConsumer, contextRegex: Regex? = null) {
+            traceEventConsumers.remove(traceEventConsumer, contextRegex)
         }
 
         /**
          * Remove all event consumers.
+         *
+         * @param contextRegex If this is `null`, all trace event consumers will be removed. Otherwise,
+         * tarce event consumer for specific contexts are removed.
          */
-        fun removeAllTraceEventConsumers() {
+        fun removeAllTraceEventConsumers(contextRegex: Regex? = null) {
             traceEventConsumers.all().asSequence().forEach { removeTraceEventConsumer(it) }
```

Comment:
Maybe I'm wrong, but should this not be like this?:

`traceEventConsumers.all().asSequence().forEach { removeTraceEventConsumer(it, contextRegex) }`

Reply:
Good catch. Fixed.